Utilities/Logging.cs

```diff
@@ -145,65 +112,109 @@ public static void incrementLogs()
             }
         }
 
-        internal static void Report(ErrorLevel errorLevel, string message, object data = null, [CallerMemberName] string memberName = "", [CallerFilePath] string filePath = "")
+        private static void Report(ErrorLevel errorLevel, string message, object data = null, [CallerMemberName] string memberName = "", [CallerFilePath] string filePath = "")
         {
-            Dictionary<string, object> thisData = PrepRollbarData(message, ref data);
+            message = Redaction.RedactEnvironmentVariables(message);
+            Dictionary<string, object> thisData = PrepRollbarData(ref data);
             if (thisData != null)
             {
                 var rollbarReport = System.Threading.Tasks.Task.Run(() => SendToRollbar(errorLevel, message, data, thisData, memberName, filePath));
             }
         }
 
-        private static Dictionary<string, object> PrepRollbarData(string message, ref object data)
+        private static Dictionary<string, object> PrepRollbarData(ref object data)
         {
             try
             {
-                // It's not possible to scrub filepaths from exception messages, so since we don't want  
-                // to collect this personal data these exceptions need to be handled locally only.
-                if (data is Exception ex)
+                // Serialize the data to a string 
+                var serialized = JsonConvert.SerializeObject(data);
+                // Redact all environment variables we find
+                serialized = Redaction.RedactEnvironmentVariables(serialized);
+                // Remove undesirable properties
+                serialized = FilterProperties(serialized);
+
+                try
                 {
-                    if (ex.Message.Contains(Constants.DATA_DIR))
-                    {
-                        return null;
-                    }
+                    data = JsonConvert.DeserializeObject<Dictionary<string, object>>(serialized);
                 }
-                else if (!(data is Dictionary<string, object>))
+                catch (Exception)
                 {
-                    var wrappedData = new Dictionary<string, object>()
+                    if (data is Exception ex)
                     {
-                        {"data", data}
-                    };
-                    data = wrappedData;
+                        data = new Dictionary<string, object>()
```

Comment:
If we're in the catch block then serializing and deserializing failed somehow, so we are back to working with the unredacted data.

Reply:
Fixed.